luigi_pipeline/lib/hail_tasks.py

```diff
@@ -72,6 +81,63 @@ def sample_type_stats(mt, genome_version, threshold=0.3):
             ht_stats['match'] = (ht_stats['matched_count']/ht_stats['total_count']) >= threshold
         return stats
 
+    @staticmethod
+    def subset_samples(mt, subset_path):
+        """
+        Subset the MatrixTable to the provided list of samples
+        :param mt: MatrixTable from VCF
+        :param subset_path: Path to a file with a single column 's'
+        :return: MatrixTable subsetted to list of samples
+        """
+        subset_ht = hl.import_table(subset_path, no_header=False).key_by('s')
+        anti_join_ht = subset_ht.anti_join(mt.cols())
+        anti_join_ht_count = anti_join_ht.count()
+        mt_sample_count = mt.cols().count()
+        missing_samples = anti_join_ht.s.collect()
```

Comment:
Can this line be moved under the `if anti_join_ht_count != 0` condition so that it is only run if `missing_samples` is used?

Same comment for `remap_sample_ids`.

Reply:
👍 Also a good idea to keep vars declaration closest to where they're used to limit scope.

Reply:
Another thing you can do is skip creating `anti_join_ht` var. Go directly to `missing_samples`.

```
missing_samples = subset_ht.anti_join(mt.cols()).collect()
```
Then whenever you want the count, use `len(missing_samples)`, which is constant time. When there is no anti-join set, the `collect` call should be minimal. When there is an anti-join set, we save on calling `count()` and `collect()`, and it's more succint.